client/layout/masterbar/logged-in.jsx

```diff
@@ -557,13 +559,15 @@ export default connect(
 		// by the user yet
 		const currentSelectedSiteId = getSelectedSiteId( state );
 		const siteId = currentSelectedSiteId || getPrimarySiteId( state );
+		const sitePlanSlug = getSitePlanSlug( state, siteId );
```

Comment:
Do we mean to use `currentSelectedSiteId` or `siteId` here?

```suggestion
		const sitePlanSlug = getSitePlanSlug( state, currentSelectedSiteId );
```

The thing is that:

* If we use `siteId`, then it will fallback to the primary site of the user;
* If we use `currentSelectedSiteId`, it will use only the current site of the user.

This means that for screens that don't have a selected site (e.g. `/stats/day`) we'll determine whether to show the Write button based on whether the primary site has an e-commerce plan or not. 

The other pieces of the condition use both - `isMigrationInProgress` uses `currentSelectedSiteId` and `domainOnlySite` uses `siteId`, so it's up to you how you want this to behave.

Reply:
Oh, nice catch there @tyxla! 🙇‍♂️ 

Based on what you said, I was definitely going for the `currentSelectedSiteId`. I'm aiming to hide the "Write" button _only_ when viewing/selecting an Ecommerce site. 

However, when I tested this, it seemed to be working! If the `siteId` is pointing to the user's primary site, then (given that my primary site is a Simple site) how it worked in the first place? 🤔

Reply:
`siteId` will be pointing to the primary site ID only if you're testing a route that doesn't have a selected site ID (like `/stats/day`. You were likely testing a route that has a site ID (`/stats/day/:siteSlug`).

Reply:
Looking at [this line](https://github.com/Automattic/wp-calypso/pull/69077/files/e589c48f920fe45aa6496d38e72a1a5ff180cf5a#diff-27374a70c13059756839dabb3fedf4c59c75444f93497261a45b2804e989cc89L558) seems that the `siteId` gets initialized based on the current one. So, that's why it worked! I get it now. 

I believe the best option for us is to choose `siteId` to better handle the initial state.

Thanks for the review, @tyxla!

Reply:
Sure thing! Thanks for considering both options. Feel free to ship either way 😉